File: backend/ModelService/Model/MastersModel/BannerModel.cs
using ModelService.CommonModel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace ModelService.Model.MastersModel
{
    public class BannerModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string TitleHindi { get; set; }
        public string Url { get; set; }
        public string AttachmentUrl { get; set; }
        public bool IsAdvt { get; set; } = false;
        public int DisplayOrder { get; set; }
        [JsonIgnore]
        public int UserId { get; set; }
    }
    public class BannerViewModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string? TitleHindi { get; set; }
        public string? Url { get; set; }
        public string? AttachmentUrl { get; set; }
        public bool? IsAdvt { get; set; } = false;
        public int DisplayOrder { get; set; }

        [JsonIgnore]
        public int TotalRows { get; set; }
    }

    public class BannerViewListModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string? TitleHindi { get; set; }
        public string? Url { get; set; }
        public string? AttachmentUrl { get; set; }
        public bool IsAdvt { get; set; }
        public int DisplayOrder { get; set; }

        public string? ModifiedByName { get; set; }
        public DateTime ModifiedDate { get; set; }
        public bool IsActive { get; set; }
        [JsonIgnore]
        public int TotalRows { get; set; }
    }
    public class BannerFilterModel : IndexModel
    {
        public string? Title { get; set; } = string.Empty;
        public int IsAdvt { get; set; }
        public int IsActive { get; set; }
        public string? FromDate { get; set; }
        public string? ToDate { get; set; }
    }
}

File: admin/src/app/core/models/master-models/banner-model.ts
import { indexModel } from "../core.model";

export class BannerModel {
	id: number;
	title: string;
	titleHindi: string | null;
	url: string | null;
	isAdvt: boolean = false;
	attachmentUrl: string | null;
	displayOrder: number;
	constructor() {
		this.isAdvt = false;
	}
}

export class BannerViewModel {
	id: number;
	title: string;
	titleHindi: string | null;
	url: string | null;
	attachmentUrl: string | null;
	isAdvt: boolean;
	displayOrder: number;
}

export class BannerViewListModel {
	id: number;
	title: string;
	titleHindi: string | null;
	url: string | null;
	isAdvt: boolean;
	attachmentUrl: string;
	displayOrder: number;
	modifiedByName: string | null;
	modifiedDate: string;
	isActive: boolean;
}

export class BannerFilterModel extends indexModel {
	title: string;
	isAdvt: number;
	fromDate: string | null;
	toDate: string | null;
}